### ISSUE TITLE
Update async code to use QFuture tasks instead of TaskWatcher

### ISSUE THREAD

**Issue by ldanzinger** [Esri/dynamic-situational-awareness-qt#368]
Asynchronous methods that returned a TaskWatcher now have new versions that return a QFuture. The new syntax allows for many new benefits, such as chaining with `.then`. With 200.3, the TaskWatcher methods will be deprecated, so we need to start migrating app code to use the new QFuture versions

**Comment by kerryrobinson** [comment]
All warnings should be fixed now, in review.

**Comment by kerryrobinson** [comment]
Kerry to verify that there are no deprecation warnings.

**Comment by JamesMBallard** [comment]
This is more of a build thing. I trust @dtedder to be able to determine if there are no more deprecation warnings with the build. I don't think Kerry needs to verify this one. Just my 2 cents.

**Comment by kerryrobinson** [comment]
Agreed.  FYI I did get some TaskWatcher warnings, but all relating to the Qt SDK deployment, not the DSA code, i.e.:
![Image](https://github.com/user-attachments/assets/6835cd7d-2c59-48a8-9032-f7121268ae76)

If this is expected, we can close this issue.

**Comment by JamesMBallard** [comment]
You should not be seeing those anymore, but it looks like indirect inclusion on TaskWatcher.h and it's not used anywhere. @dtedder can you check to see how TaskWatcher is getting included? It should be forward declared everywhere in our SDK headers, and if it's unused then TaskWatcher.h should never be included.

It's possible there are no issues here, but I am curious to figure out why these are popping up. We wouldn't want our users to see a flood of warnings like this for classes they aren't using.

**Comment by dtedder** [comment]
Maybe here?
https://github.com/Esri/dynamic-situational-awareness-qt/blob/01c0e13df0a4ed1a7e90b4d7ec80e21741a123ff/Shared/pch.hpp#L37

The only other references I can find are from the Toolkit which still includes it in a few files.

**Comment by JamesMBallard** [comment]
Yes that's it. That should be removed now. The toolkit has been updated to no longer use TaskWatcher. @dtedder can you make that tweak to pch.hpp?
